webpack/assets/javascripts/react_app/components/common/TypeAheadSelect/TypeAheadSelectReducer.js

```diff
@@ -0,0 +1,34 @@
+import Immutable from 'seamless-immutable';
+import {
+  INIT,
+  UPDATE_OPTIONS,
+  UPDATE_SELECTED,
+} from './TypeAheadSelectConstants';
+
+const initialState = Immutable({});
+
+export default (
+  state = initialState,
+  { type, payload: { id, options, selected } = {} }
+) => {
+  switch (type) {
+    case INIT:
+      return state.setIn([id], {
+        ...state[id],
+        options,
+        selected,
+      });
+    case UPDATE_OPTIONS:
+      return state.setIn([id], {
+        ...state[id],
+        options,
+      });
+    case UPDATE_SELECTED:
+      return state.setIn([id], {
+        ...state[id],
+        selected: selected.map(item => (item.label ? item.label : item)),
```

Comment:
Please elaborate on the request to move this mapping into a selector - I don't understand why :)

Reply:
It is just a convention, to move that kind of logic into a selector,
And keep the component, action, reducers logic simple, plus you could handle more edge cases there too.